operator/controllers/keptntask/job_utils.go

```diff
@@ -0,0 +1,115 @@
+package keptntask
+
+import (
+	"context"
+	"fmt"
+	"github.com/imdario/mergo"
+	klcv1alpha1 "github.com/keptn-sandbox/lifecycle-controller/operator/api/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func (r *KeptnTaskReconciler) createJob(ctx context.Context, req ctrl.Request, task *klcv1alpha1.KeptnTask) error {
+	job := ""
+	definition, err := r.getTaskDefinition(ctx, task.Spec.TaskDefinition, req.Namespace)
+	if err != nil {
+		r.Recorder.Event(task, "Warning", "TaskDefinitionNotFound", fmt.Sprintf("Could not find KeptnTaskDefinition / Namespace: %s, Name: %s ", task.Namespace, task.Spec.TaskDefinition))
+		return err
+	}
+
+	if !reflect.DeepEqual(definition.Spec.Function, klcv1alpha1.FunctionSpec{}) {
+		job, err = r.createFunctionJob(ctx, req, task, definition)
+		if err != nil {
+			return err
+		}
+	}
+
+	task.Status.JobName = job
+	task.Status.Status = klcv1alpha1.TaskPending
+	err = r.Client.Status().Update(ctx, task)
+	if err != nil {
+		r.Log.Error(err, "could not update configmap status reference for: "+definition.Name)
+	}
+	r.Log.Info("updated configmap status reference for: " + definition.Name)
+	return nil
+}
+
+func (r *KeptnTaskReconciler) createFunctionJob(ctx context.Context, req ctrl.Request, task *klcv1alpha1.KeptnTask, definition *klcv1alpha1.KeptnTaskDefinition) (string, error) {
+	params, hasParent, err := parseFunctionTaskDefinition(definition)
+	var parentJobParams FunctionExecutionParams
+	if err != nil {
+		return "", err
+	}
+	if hasParent {
+		parentDefinition, err := r.getTaskDefinition(ctx, definition.Spec.Function.FunctionReference.Name, req.Namespace)
+		if err != nil {
+			r.Recorder.Event(task, "Warning", "TaskDefinitionNotFound", fmt.Sprintf("Could not find KeptnTaskDefinition / Namespace: %s, Name: %s ", task.Namespace, task.Spec.TaskDefinition))
+			return "", err
+		}
+		parentJobParams, _, err = parseFunctionTaskDefinition(parentDefinition)
+		if err != nil {
+			return "", err
+		}
+		err = mergo.Merge(&params, parentJobParams)
+		if err != nil {
+			r.Recorder.Event(task, "Warning", "TaskDefinitionMergeFailure", fmt.Sprintf("Could not merge KeptnTaskDefinition / Namespace: %s, Name: %s ", task.Namespace, task.Spec.TaskDefinition))
+			return "", err
+		}
```

Comment:
The secure parameters of the parent are left out on purpose?

Reply:
they are included as in the parent-child case, the whole task definition is merged